AudioGenerator.py

```diff
@@ -0,0 +1,181 @@
+        if sampleValue > 1:
+            sampleValue = 1
+        elif sampleValue < -1:
+            sampleValue = -1
+        sinWave.append(sampleValue)
+    return sinWave
+
+def makeNoise (numSample, maxAmp):
+    noiseList = []
+    for index in range(0, numSample):
+        sample = random.randint(maxAmp * -1, maxAmp) / float(maxAmp)
+        noiseList.append(sample)
+    return noiseList
+
+def writeToFile(sinSound, fileName, numChannels, maxAmp):
+    struct.pack('h', 1000)
+    file = wave.open(str(fileName) + '.wav', 'w')
+    file.setparams((numChannels, 2, audioHz, len(sinSound), 'NONE', 'not compressed'))
+    for sample in sinSound:                                            # converts sin wave to  16 bit depth audio values
+        sample = int(sample*maxAmp)
+        data = ''.join(''.join(struct.pack('h', sample)))              # .join allows me to put the list into a string
+        file.writeframes(data)
+    file.close()
+    return str(fileName)
+
+#finds and plays a file in your OS's default media player
+def playFile(fileName):
+    fileLocation = str(__file__)
+    fileLocation = fileLocation.replace('AudioGenerator.py', str(fileName) + '.wav', 1)
+    os.system('start ' + fileLocation)
+    return fileLocation
+
+def mergeSound(list1, list2, startPoint):                              # list 1 should be the larger sin list
+    for sample in range(startPoint, len(list1)):                       # currently has a bug where you must merge
+        if sample < len(list2):                                        #  with a sin wave for the code to work
+            newSample = list1[sample] + list2[sample]
+            newSample = newSample / 2.0
+            list1[sample] = newSample
+        else:
+            break
+    return list1
+
+def loadFile(fileName):                                                # Must be mono and have a 16 bit depth
+    file = wave.open(fileName + '.wav', 'r')
+    sampleList = []
+    numSamples = file.getnframes()
+    for sample in range(0, numSamples):
+        rawData = file.readframes(1)
+        data = struct.unpack('<h', rawData)
+        sampleList.append(data[0]/32767.0)
+    return sampleList
+
+def stringSound(sound1,sound2):                                        # strings one sound onto the end of another
+    for sample in range(0, len(sound2)):
+        sound1.append(sound2[sample])
+    return sound1
+
+def echoSound(soundList, numEcho, sampleGap, volumeDecrease):
+    for echo in range(1, numEcho + 1):
+        sampleDifference = (echo + 1) * sampleGap
+        for sample in range(sampleDifference, len(soundList)):
+                newSample = soundList[sample] + soundList[sample - sampleDifference]
+                volumeChange = volumeDecrease * echo
+                if volumeChange <= 1 and volumeChange >= 0:
+                    newSample = (newSample * volumeChange) / 2
+                    soundList[sample] = newSample
+    return soundList
+
+class Melody:
```

Comment:
makeSin and stringSound are repeated above. The only apparent differences in the methods of this class is that they use fields. The code can be structured differently to avoid repeating code.